packages/lib-classifier/src/store/Workflow/Workflow.js

```diff
@@ -35,12 +36,13 @@ const Workflow = types
   }))
 
   .actions(self => {
-    function selectSubjectSet(id) {
-      self.activeSubjectSet = id
+    function * selectSubjectSet(id) {
+      yield self.subjectSets.setActive(id)
+      return self.subjectSets.active
```

Comment:
Should this have a try...catch? What if this fails?

Reply:
Maybe it should be wrapped in `tryReference`?

Reply:
Yeah, that sounds good and maybe a test for this possibility? What do we expect if this is undefined?

Reply:
Return null or undefined I guess. I can't remember which is returned by tryReference for an invalid reference. At the moment, I don't think the return value is used. It's there because MST actions can't handle async/await.

Reply:
Looking at the tests, they need to be updated to use `async` anyway, since `selectSubjectSet` now returns a promise. And the API client needs to be mocked in the tests. And I've broken the Workflow factory with the changes in this PR.

I'm working on that this morning.

Reply:
I've completely rewritten the tests now so the async functions should be tested correctly and cases where `subjectSets.fetchResource(id)` returns undefined are included.